File: algorithms/2_medium/M/01785_minimum_elements_to_add_to_form_a_given_sum/solution.c
int minElements(const int *restrict nums, const int len, const int limit, const int goal) {
    int64_t sum = 0;
    for (int i = 0; i < len; i += 1) {
        sum += nums[i];
    }
    const int64_t diff = ABS((int64_t)goal - sum);

    return (int)((diff + limit - 1) / limit);
}

File: algorithms/2_medium/M/01785_minimum_elements_to_add_to_form_a_given_sum/solution.rs
impl Solution {
    pub fn min_elements(nums: Vec<i32>, limit: i32, goal: i32) -> i32 {
        let (goal, limit) = (goal as i64, limit as i64);
        let sum: i64 = nums.iter().map(|x| *x as i64).sum();
        let diff = (goal - sum).abs();

        ((diff + limit - 1) / limit) as i32
    }
}
